CMakeLists.txt

```diff
@@ -264,6 +264,28 @@ foreach(RELTYP RELEASE DEBUG RELWITHDEBINFO MINSIZEREL)
         endforeach()
 endforeach()
 
+#
+# Handling of LTO/IPO
+#
+
+option(DISABLE_LTO "Disable use of LTO when building. Defaults to being enabled if supported." FALSE)
+
+if(NOT DISABLE_LTO)
+        include(CheckIPOSupported)
+
+        message(CHECK_START "Checking for LTO support")
+        check_ipo_supported(RESULT HAVE_LTO)
+
+        if(HAVE_LTO)
+                message(CHECK_PASS "supported")
+                set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+        else()
+                message(CHECK_FAIL "not supported")
+        endif()
+else()
+        message(STATUS "Not checking for LTO support, as it has been explicitly disabled")
```

Comment:
IMHO, there's no reason to print this.

Reply:
The intent was to be explicit about this in the output to simplify debugging when users post build logs. By saying that _we_ are not checking for it, we then know that any LTO flags that happen to be in the compiler/linker flags in the build log are there because the user (or their build environment) added those flags in CFLAGS/CXXFLAGS/LDFLAGS or somewhere similar.

Reply:
I don't think we should take into consideration such cases when maintaining CMake code, because it over-complicates things for no good (enough) reason. In any case, I understand your rationale so it's fine to ignore this.